lib/raxx.ex

```diff
@@ -791,13 +801,24 @@ defmodule Raxx do
   Complication with such functionality are discussed here - https://github.com/phoenixframework/phoenix/pull/1402
   Sinatra has a very complete test suite including a back implementation - https://github.com/sinatra/sinatra/blob/9bd0d40229f76ff60d81c01ad2f4b1a8e6f31e05/test/helpers_test.rb#L183
   """
+  @spec redirect(String.t(), status: status, body: body, content_type: String.t()) ::
```

Comment:
Does this spec assume all of these keys are optional?

Reply:
Yes, according to that spec all keys are optional. Also the order does not matter.

This is a white-list, so dialyzer will fail on e.g. `redirect(url, bogus: :foo)`

Reply:
I was unsure after you asked, so I made a small test:

```elixir
  @spec my_fun(a: integer, b: integer) :: nil
  def my_fun(_opts) do
  end

  def other_fun do
    my_fun(a: 1, b: 2)
    my_fun(a: 1)
    my_fun(b: 2, a: 1)
    my_fun(a: 1, c: 5)
  end
```

That last line makes dialyzer fail :)